src/colmap/estimators/cost_functions.h

```diff
@@ -134,6 +134,62 @@ class ReprojErrorConstantPoseCostFunction {
   const double observed_y_;
 };
 
+// Bundle adjustment cost function for variable
+// camera pose and calibration parameters, and fixed point.
+template <typename CameraModel>
+class ReprojErrorConstantPoint3DCostFunction {
+ public:
+  explicit ReprojErrorConstantPoint3DCostFunction(
+      const Eigen::Vector2d& point2D, const Eigen::Vector3d& point3D)
+      : observed_x_(point2D(0)),
+        observed_y_(point2D(1)),
+        point_x_(point3D(0)),
+        point_y_(point3D(1)),
+        point_z_(point3D(2)) {}
+
+  static ceres::CostFunction* Create(const Eigen::Vector2d& point2D,
+                                     const Eigen::Vector3d& point3D) {
+    return (new ceres::AutoDiffCostFunction<
+            ReprojErrorConstantPoint3DCostFunction<CameraModel>,
+            2,
+            4,
+            3,
+            CameraModel::kNumParams>(
+        new ReprojErrorConstantPoint3DCostFunction(point2D, point3D)));
+  }
+
+  template <typename T>
+  bool operator()(const T* const cam_from_world_rotation,
+                  const T* const cam_from_world_translation,
+                  const T* const camera_params,
+                  T* residuals) const {
+    Eigen::Matrix<T, 3, 1> point3D;
+    point3D[0] = T(point_x_);
+    point3D[1] = T(point_y_);
+    point3D[2] = T(point_z_);
```

Comment:
```suggestion
    const Eigen::Matrix<T, 3, 1> point3D = {T(point_x_), T(point_y_), T(point_z_)};
```

Reply:
Only some versions of Eigen support std initializer list.
https://gitlab.com/libeigen/eigen/-/issues/2137